my-dsl.md

```diff
@@ -1,39 +1,38 @@
 # Language
 
-_What is the name of the language? Link the name to its webpage
-(if appropriate)._
+[Csound](https://csound.com/)
 
 # Domain
 
-_Describe the language's domain in five words._
+Audio synthesis with real-time inputs
 
 # Computational model
 
-_We don't yet have a great definition of the term "computational model".
-For now, try to come up with the clearest, most concise explanation of
-what happens when a program in your DSL runs._
+Csound code contains an instrument section and a score section.
+The instrument section defines parameters of each different instrument used (constant throughout the program runtime), and the score section defines when different events occur after initial execution.
+Thus, Csound can be used as a sequencer to synthesize audio patterns.
+However, the program can also read real-time inputs from devices such as MIDI keyboards to play sounds and apply effects at times defined during the program runtime.
 
 # DSL-ness
 
-_Fowler writes about a spectrum of languages, from general-purpose languages to
-"purely" domain-specific. Where does the DSL you chose fall on this spectrum,
-and why?_
+I would say this is more on the domain-specific side of the spectrum even edging a bit past since it has very limited functionality (I don't think you can even create a "hello world" program since there are no strings).
+The code is mostly definitions with limited freedom on functions that can be applied.
 
 # Internal or external?
 
-_Is the language implemented as an internal or external DSL?
-Justify your answer._
+Csound is an external programming language because the syntax is uninterpretable in C.
+It is more reminiscent of an XML file with information that could be passed to a C program, but it does have more syntactic structure (including loops and arithmetic operations).
```

Comment:
I also reflected on a similar aspect of when analyzing my DSL, namely the fact the DSL seemed relatively limited but had loops and arithmetic operations. 

This comment helps me understand one of the questions that I raised when doing the assignment. Does the presence of loops make a program Turing complete? In the Fowler text, I noted that "DSLs usually avoid the regular imperative control structures (conditionals and loops)." However, I did not really think about the domain focus which you did a great job touching on.